### PR TITLE
Track JS widgets in GA4 (#13238)

### PR DESCRIPTION
**Pull request by stephaniehobson**
## One-line summary

Track JS widgets in GA4 with `widget_action`

## Significant changes and points to review

We currently track JS widgets using a mix of `in-page-interaction`, `non-interaction` and the old UA category/action/label to track user interactions with JS widgets on the page.

This is a generic event meant as a catch-all for existing and future interactions with JS widgets that are not conversions.

- Do the event name, properties, and how to use it make sense?
- Have all the appropriate events been instrumented?
- Is the documentation sufficient?
- Basically, before we're stuck with this forever is there anything that is going to be confusing or annoying?

## Issue / Bugzilla link

https://github.com/mozilla/bedrock/issues/13238

## Testing

Files can be viewed locally to see what is being written to the DataLayer.

### CODE REVIEW

File: media/js/firefox/all/all-downloads-unified-init.es6.js [hunk 1]
```diff
@@ -25,11 +25,19 @@ import MzpModal from '@mozilla-protocol/core/protocol/js/modal';
...
                 event: 'in-page-interaction',
                 eAction: 'link click',
                 eLabel: eventLabel
             });
+            // GA4
+            window.dataLayer.push({
+                event: 'widget_action',
+                type: 'modal',
+                action: 'open',
+                name: eventLabel
```
Comment (stephaniehobson) [inline media/js/firefox/all/all-downloads-unified-init.es6.js:39]: eventLabel could be one of: 'Get Browser Help', 'Get Installer Help' but neither is displayed to the user. I waffled on if these should be the name or the label.

File: media/js/firefox/challenge-the-default/challenge-the-default.es6.js [hunk 2]
```diff
@@ -79,10 +88,20 @@ kittenButton.addEventListener(
...
             eAction: 'Kitten modal'
         });
+
+        // GA4
+        window.dataLayer.push({
+            event: 'widget_action',
+            type: 'easter egg',
+            action: 'discover',
+            name: 'kitten modal'
+        });
```
Comment (alexgibson) [inline media/js/firefox/challenge-the-default/challenge-the-default.es6.js:104]: `type: 'easter egg'` and `action: 'discover'` feels a bit too one-off maybe? 

I wonder if this one could be a more generic modal type event? Not sure what to do with the others though :confused:

File: docs/attribution/0001-analytics.rst [hunk 3]
```diff
@@ -339,6 +364,97 @@ There are two ways to use TrackProductDownload:
...
+   * - It can perform several actions.
+
+       *(A modal can be opened and closed.)*
+     - It does one action.
+   * - There could be several on the page doing different things.
+
+       *(An accordion list of FAQs)*
+     - There could be several on the page doing the same thing.
+
+       *(A download button in the header and footer.)*
```
Comment (alexgibson) [inline docs/attribution/0001-analytics.rst:396]: Not something that needs changing necessarily, but I find the source order of this table pretty hard to follow?

**Review by alexgibson — APPROVED** [review]
r+wc
